File: CSharp/LeetCode/Problem560.cs
/// <summary>
/// https://leetcode.com/problems/subarray-sum-equals-k/
/// 
/// Given an array of integers nums and an integer k, return the total number of continuous subarrays whose 
/// sum equals to k.
/// </summary>

namespace Problem560
{
    public class Solution
    {
        public int SubarraySum(int[] nums, int k)
        {
            var result = 0;

            var sumFromBeginning = new int[nums.Length + 1];
            sumFromBeginning[0] = 0;
            for (var i = 1; i < nums.Length + 1; i++)
                sumFromBeginning[i] = sumFromBeginning[i - 1] + nums[i - 1];

            for (var i = 0; i < sumFromBeginning.Length; i++)
            for (var j = i + 1; j < sumFromBeginning.Length; j++)
                if (sumFromBeginning[j] - sumFromBeginning[i] == k)
                    result++;

            return result;
        }
    }
}

File: Problem560.py
# https://leetcode.com/problems/subarray-sum-equals-k/

# Given an array of integers nums and an integer k, return the total number of continuous subarrays whose
# sum equals to k
from typing import List


class Solution:
    def subarraySum (self, nums: List[int], k: int) -> int:
        sums = []
        for i in range(0, len(nums)):
            sums.append((sums[i-1] if i > 0 else 0) + nums[i])

        result = 0
        for end in range(0, len(sums)):
            for start in range(0, end):
                interval_sum = sums[end] - sums[start]
                if interval_sum == k:
                    result += 1

        return result
